doc/develop/languages/c/index.rst

```diff
@@ -63,6 +63,19 @@ application.
 
 .. _`C Standard Library`: https://en.wikipedia.org/wiki/C_standard_library
 
+.. _c_library_formatted_output:
+
+Formatted Output
+****************
+
+C defines standard formatted output functions such as :c:func:`printf` and
+:c:func:`sprintf` and these functions are implemented by the C standard
```

Comment:
afaik we don't have Doxygen entries for `printf`, `sprintf`, etc., so `:c:func:` does nothing here (and will actually generate a warning in nitpicky mode). I'd suggest using literals instead.

Reply:
But we can add Doxygen entries for these in the future and, if we just use literals, we will have to go through the entire doc to update their references.

AFAICS, it does not cost us anything to keep `:c:func:` (or `:kconfig:option` for that matter), so why not keep it?

Reply:
the purpose of `:c:func:` is to link to an existing `.. c:func:` directive, not to indicate this is a C function. Sphinx nitpicky mode (not used nowadays due to the amount of issues we have) will raise a warning if using roles in this way.